site/src/pages/TaskPage/TaskApps.tsx

```diff
@@ -17,74 +18,68 @@ import { type FC, useState } from "react";
 import { Link as RouterLink } from "react-router";
 import { cn } from "utils/cn";
 import { docs } from "utils/docs";
+import { getAgents } from "utils/workspace";
 import { TaskAppIFrame } from "./TaskAppIframe";
 
 type TaskAppsProps = {
 	task: Task;
 };
 
-type AppWithAgent = {
-	app: WorkspaceApp;
+type WorkspaceAppWithAgent = WorkspaceApp & {
 	agent: WorkspaceAgent;
 };
 
 export const TaskApps: FC<TaskAppsProps> = ({ task }) => {
-	const agents = task.workspace.latest_build.resources
-		.flatMap((r) => r.agents)
-		.filter((a) => !!a);
-
-	// The Chat UI app will be displayed in the sidebar, so we don't want to show
-	// it here
-	const apps = agents
+	const agents = getAgents(task.workspace);
+	const apps: WorkspaceAppWithAgent[] = agents
 		.flatMap((agent) =>
 			agent.apps.map((app) => ({
-				app,
+				...app,
 				agent,
 			})),
 		)
+		// The Chat UI app will be displayed in the sidebar, so we don't want to
+		// show it as a tab.
 		.filter(
-			({ app }) =>
-				!!app && app.id !== task.workspace.latest_build.ai_task_sidebar_app_id,
+			(app) => app.id !== task.workspace.latest_build.ai_task_sidebar_app_id,
 		);
-
-	const embeddedApps = apps.filter(({ app }) => !app.external);
-	const externalApps = apps.filter(({ app }) => app.external);
-
-	const [activeAppId, setActiveAppId] = useState<string | undefined>(
-		embeddedApps[0]?.app.id,
-	);
+	const embedApps = apps.filter((app) => !app.external);
+	const externalApps = apps.filter((app) => app.external);
```

Comment:
I know it was already like this, but fact that these groups are perfect partitions makes iterating over the list twice feel a little bad. I'd probably do something like

```
const embeddedApps = [];
const externalApps = [];

for (const app of apps) {
  if (app.external) {
    externalApps.push(app);
  } else {
    embeddedApps.push(app);  
  }
}